src/main/java/ch/alpine/tensor/SparseArray.java

```diff
@@ -9,51 +9,22 @@
 import java.util.NavigableMap;
 import java.util.Objects;
 import java.util.TreeMap;
-import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
-import ch.alpine.tensor.alg.Array;
-import ch.alpine.tensor.alg.Dimensions;
 import ch.alpine.tensor.ext.Integers;
 
-/** API EXPERIMENTAL
- * 
- * WIP */
 /* package */ class SparseArray extends AbstractTensor implements Serializable {
-  /** @param tensor
-   * @param fallback
-   * @return
-   * @throws Exception if given tensor does not have array structure */
-  public static Tensor of(Tensor tensor, Scalar fallback) {
-    Dimensions dimensions = new Dimensions(tensor);
-    if (dimensions.isArray()) {
-      List<Integer> size = dimensions.list();
-      if (0 == size.size())
-        return tensor;
-      SparseArray sparseArray = new SparseArray(size, fallback);
-      Array.forEach(list -> {
-        Tensor entry = tensor.get(list); // entry is scalar due to dimension check above
-        if (!fallback.equals(entry))
-          sparseArray.set(entry, list);
-      }, size);
-      return sparseArray;
-    }
-    throw TensorRuntimeException.of(tensor);
-  }
-
-  // ---
   private final List<Integer> size;
   private final Scalar fallback;
   private final NavigableMap<Integer, Tensor> navigableMap;
```

Comment:
q: I've taken a `Trie<int[], IExpr` implementation in the Symja project:
- https://github.com/axkr/symja_android_library/blob/6e8e77c1cd81ec35f7eb76bae0848fc3a8fe1930/symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/expression/data/SparseArrayExpr.java#L49
 
What is the reason (or benefit) for choosing a NavigableMap here?

Reply:
Hi Axel, danke fuer Deine Frage/Anmerkung! Ich habe das SparseArray gestern erst mal zum Ausprobieren programmiert. Es ist noch sehr experimentell. (Mir ist zB aufgefallen, dass Transpose fuer SparseArrays separat implementiert werden muss.) Du hast Recht bzgl der Map. Man koennte (und sollte) einfach eine HashMap nehmen. Ich werde das aendern. Wahrscheinlich sollte ich statt einem int auch gleich einen multiindex nehmen (so wie bei Dir). Uebrigens, in diesem Branch findest Du auch eine deutliche Vereinfachung von der Implementation von Transpose (falls das noch fuer Dich interessant ist). Viele Gruesse, Jan

Reply:
@axkr ich sehe gerade dass Du auf `Trie` verwiesen hast. das kenne ich nicht, und muss es mir noch erst anschauen.

Reply:
mir faellt auf, dass man fuer tensor::extract und tensor::block eine subMap braucht, daher das NavigableMap...

Reply:
In classic implementations you will use CSR structure:
- https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_row_(CSR,_CRS_or_Yale_format)

but I've choosen a trie because it's easier to implement:
- https://en.wikipedia.org/wiki/Trie

Reply:
danke fuer die nuetzlichen Hinweise! ein Thema, das ich noch interessant finde, ist SparseArray's mit Symmetrien, bzw Antisymmetrien. zB Regeln wie A(i,j,k) == A(j,i,k).

Reply:
@axkr nachdem ich mir Trie auf Wikipedia angeschaut habe, bekomme ich den Eindruck, dass die Trie Datenstruktur equivalent ist zu 

    Map<Integer, Map<Integer, Map<Integer, ... , T>>>

Koennte es sein, dass unsere Implementationen speicher- und effizienz-maessig equivalent sind?